controllers/configmap_controller.go

```diff
@@ -142,6 +142,16 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 	// At this point configMap will be set properly
 	switch req.NamespacedName.Name {
 	case servicescm.Name:
+		// If a ConfigMap with invalid values is found, WMCO will delete and recreate it with proper values
```

Comment:
What is the source of truth of the ConfigMap? At the moment the ConfigMap is empty but if I manually create a ConfigMap with some entries as `Parse()` excepts then that ConfigMap will persist. You need stronger validation for invalid data here.

Reply:
WMCO is the source of truth. In my view, we have sufficient effort to ensure this, as `Parse()` also does service validation. If someone were to create a configmap adhering to the versioned resource naming structure and all data schema requirements, within our operator namespace while the operator is not running, IMO that's on the user.

If your ask is to have a `requiredServices` and `requiredFiles` set to check against the ConfigMap's data, I think that should be implemented when we start moving services into WICD's scope.